datadog/resource_datadog_synthetics_check.go

```diff
@@ -0,0 +1,272 @@
+// For more info about writing custom provider: shttps://www.terraform.io/docs/extend/writing-custom-providers.html
+
+package datadog
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	datadog "github.com/zorkian/go-datadog-api"
+)
+
+// TODO: Add support for browser tests
+var syntheticsTypes = []string{"api"}
+
+func resourceDatadogSyntheticsTest() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceDatadogSyntheticsTestCreate,
+		Read:   resourceDatadogSyntheticsTestRead,
+		Update: resourceDatadogSyntheticsTestUpdate,
+		Delete: resourceDatadogSyntheticsTestDelete,
+		Exists: resourceDatadogSyntheticsTestExists,
+		Schema: map[string]*schema.Schema{
+			"type": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringInSlice(syntheticsTypes, false),
+			},
+			"request": syntheticsTestRequest(),
+			"assertions": {
+				Type:     schema.TypeList,
+				Required: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeMap,
+				},
+			},
+			"locations": {
+				Type:     schema.TypeList,
+				Required: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+					// TODO: validation with regexp
+				},
+			},
+			"options": syntheticsTestOptions(),
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"message": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "",
+			},
+			"tags": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"paused": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+		},
+	}
+}
+
+func syntheticsTestRequest() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeMap,
+		Required: true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"method": {
+					Type:     schema.TypeString,
+					Required: true,
+				},
+				"url": {
+					Type:     schema.TypeString,
+					Required: true,
+				},
+			},
+		},
+	}
+}
+
+func syntheticsTestOptions() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeMap,
+		Optional: true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"tick_every": {
```

Comment:
Please add all options we support (I'm thinking about min_failure_duration and min_location_failed).